src/VirtoCommerce.SitemapsModule.Data/BackgroundJobs/SitemapProcessJob.cs

```diff
@@ -0,0 +1,235 @@
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Error exporting sitemap for store {storeId}.", store.Id); // Log error
+                }
+            }
+        }
+    }
+
+    public Task BackgroundDownload(string storeId, string baseUrl, string localTmpFolder, SitemapDownloadNotification notification)
+    {
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(storeId);
+
+        if (!Uri.TryCreate(baseUrl, UriKind.Absolute, out _))
+        {
+            throw new ArgumentException($"Incorrect base URL {baseUrl}");
+        }
+
+        return InnerBackgroundDownload(storeId, baseUrl, localTmpFolder, notification);
+    }
+
+    public Task BackgroundExportToAssets(string storeId, string baseUrl, SitemapExportToAssetNotification notification)
+    {
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(storeId);
+
+        if (!Uri.TryCreate(baseUrl, UriKind.Absolute, out _))
+        {
+            throw new ArgumentException($"Incorrect base URL {baseUrl}");
+        }
+
+        return InnerBackgroundExportToAssets(storeId, baseUrl, notification);
+    }
+
+    private async Task InnerBackgroundExportToAssets(string storeId, string baseUrl, SitemapExportToAssetNotification notification)
+    {
+        void SendNotificationWithProgressInfo(ExportImportProgressInfo c)
+        {
+            notification.Description = c.Description;
+            notification.ProcessedCount = c.ProcessedCount;
+            notification.TotalCount = c.TotalCount;
+            notification.Errors = c.Errors?.ToList() ?? [];
+
+            _notifier.Send(notification);
+        }
+
+        var outputAssetFolder = string.Format(Core.ModuleConstants.StoreAssetsOutputFolderTemplate, storeId);
+
+        try
+        {
+            var sitemapXmlBlobInfo = await ExportSitemapPartAsync(storeId, baseUrl, outputAssetFolder, ModuleConstants.SitemapFileName, SendNotificationWithProgressInfo);
+
+            foreach (var sitemapUrl in await _sitemapXmlGenerator.GetSitemapUrlsAsync(storeId, baseUrl))
+            {
+                await ExportSitemapPartAsync(storeId, baseUrl, outputAssetFolder, sitemapUrl, SendNotificationWithProgressInfo);
+            }
+
+            notification.Description = "Sitemap export to store assets finished";
+            notification.SitemapXmlUrl = sitemapXmlBlobInfo.Url;
+        }
+        catch (Exception exception)
+        {
+            notification.Description = "Sitemap export to store assets failed";
+            notification.Errors.Add(exception.ExpandExceptionMessage());
+        }
+        finally
+        {
+            notification.Finished = DateTime.UtcNow;
+            await _notifier.SendAsync(notification);
+        }
+    }
+
+    private async Task InnerBackgroundDownload(string storeId, string baseUrl, string localTmpFolder, SitemapDownloadNotification notification)
+    {
+        void SendNotificationWithProgressInfo(ExportImportProgressInfo c)
```

Comment:
InnerBackgroundExportToAssets method has the same local function.  What if we make a method that will receive the notification and return this function to us?

Reply:
Fixed